### PR TITLE
async function instead of callback function

### LINKED ISSUE #4: Return a `Promise` if no callback is provided
This would be the example from the readme with promise based:
```js
var write = require('stream-write');
var http = require('http');

http.createServer(function(req, res){
  (async function (){
    while (true) {
      const open = await write(res, ''+Math.random())
      if (!open) break
    }
  })().catch(console.error);
}).listen(8000);
```

### PR DESCRIPTION
**Pull request by ChocolateLoverRaj**
Closes #4 

Checks `stream.writable` instead of `stream.socket.writable`

Removed Makefile because I don't think it's needed

Changed tests to use baretest because tape was not working and node.js assert module already has assertion methods.

Added `pnpm-lock.yaml` to gitignore because I did `pnpm i`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,11 +3,11 @@
   "version": "1.0.1",
   "description": "Write to a stream callback style",
   "scripts": {
-    "test": "make test"
+    "test": "node test"
   },
   "repository": "juliangruber/stream-write",
   "devDependencies": {
-    "tape": "^4.6.3"
+    "baretest": "^2.0.0"
```
Comment (juliangruber) [inline package.json:10]: why did you change the test framework? 🤔

Comment (juliangruber) [reply package.json:10]: Usually I'd say just upgrade `tape` for async support, or switch to `tap`, but I'm curious to try out this new one :+1:

File: test.js [hunk 2]
```diff
@@ -97,3 +80,6 @@ test('listener cleanup', function(t){
   }
 });
 
+!(async function() {
+  await test.run()
+})()
```
Comment (juliangruber) [inline test.js:85]:
```suggestion
test.run()
```

File: index.js [hunk 3]
```diff
@@ -1,33 +1,15 @@
...
-    stream.once('finish', next);
-
-    function next(){
-      stream.removeListener('error', error);
-      stream.removeListener('drain', next);
-      stream.removeListener('finish', next);
-      cb(null, stream.writable);
-    }
+  if (!stream.write(chunk)) {
+    await Promise.race([once(stream, 'drain'), once(stream, 'finish')])
```
Comment (juliangruber) [inline index.js:11]: I love how much simpler this has become

File: index.js [hunk 4]
```diff
@@ -1,33 +1,15 @@
...
-    errored = true;
-    cb(err);
-  }
-
-  if (stream.socket && !stream.socket.writable) {
-    var err = new Error('write after end');
+module.exports = async function write(stream, chunk){
+  if (!stream.writable) {
+    var err = new Error('Stream is not writable');
     err.status = 200;
```
[the remark this answers is not in the document]
Comment (juliangruber) [reply index.js:6]: I don't recall really :/

File: index.js [hunk 5]
```diff
@@ -1,33 +1,15 @@
...
-  function error(err){
-    errored = true;
-    cb(err);
-  }
-
-  if (stream.socket && !stream.socket.writable) {
-    var err = new Error('write after end');
+module.exports = async function write(stream, chunk){
+  if (!stream.writable) {
+    var err = new Error('Stream is not writable');
```
Comment (juliangruber) [inline index.js:5]:
```suggestion
    const err = new Error('Stream is not writable');
```

Comment (juliangruber) [reply index.js:5]: and we could also just throw it in the same line

File: index.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (juliangruber) [reply index.js:6]: It must have something to do with http frameworks like express/connect/koa, who read `err.status` for constructing an error page. But not sure why it would be 200 🤔  Let's remove it

**Review by juliangruber — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (ChocolateLoverRaj) [reply package.json:10]: - Tape wasn't working when I used async functions. I didn't realize that I should update it.
- I read about baretest and its docs say it's super small and fast. One thing is that tape had assertion methods which aren't needed because we can just use node js assert.

**Review by juliangruber — CHANGES_REQUESTED** [review]
Code lgtm 🥳 ! Would you mind updating the README as well?
